src/tcgwars/logic/groovy/TcgStatics.groovy

```diff
@@ -1146,6 +1146,99 @@ class TcgStatics {
   static boolean isGxPerformed(){
     bg.em().retrieveObject("gx_"+my.owner)
   }
+
+  /* General checks for attacks and abilities */
+
+  /**
+   *
+   * Does a customized assert with an automated fail warning, looking for any Pokémon following the given filters.
+   *
+   * @param params Optional settings that can be added:
+   *   + benched: If true, checks for only Benched Pokémon; otherwise also includes the Active.
+   *   + opp: If true, checks for the opponent's bench instead of "my" bench.
+   *   + hasType: If set, restricts to benched Pokémon of a single specific type.
+   *   + hasPokemonEX/hasPokemonGX/hasPokemonV/hasPokemonMAX: Can be expanded if needed. All of these unset will have the method search for any Pokémon no matter what, but if even a single one is set true it'll only filter those that are set true as well.
+   *   + info: If set, it'll replace the end of the failed assert warning with a custom text, instead of the default "follow the stated condition(s)".
+   *   + repText: If true, params.info will override the entirety of the failed assert warning.
+   *
+   * @param filter Additional condition the filtered benched Pokémon must follow. Defaults to true (so any Pokémon).
+   *
+   */
+  static void assertAnyPokemonInPlay(params=[:], Closure filter = null) {
+    def failMessage
+    def checkedPlayer = params.opp ? opp : my
+    def checkedArea = params.benched ? checkedPlayer.benched : checkedPlayer.all
+
+    def hasPokeCnt = 0
+    [params.hasPokemonEX, params.hasPokemonGX, params.hasPokemonV, params.hasPokemonVMAX].each{hasPokeCnt += 1}
+
+    def areaFilter = (
+      (
+          !params.hasType || it.types.contains(params.hasType)
+      ) && (
+          !hasPokeCnt || (
+              (params.hasPokemonEX && it.pokemonEX) ||
+              (params.hasPokemonGX && it.pokemonGX) ||
+              (params.hasPokemonV && it.pokemonV) ||
+              (params.hasPokemonVMAX && it.pokemonVMAX)
+          )
+      ) && (
+          filter == null || filter.call(it)
+      )
+    )
+
+    if (params.info + !params.repText) {
+      failMessage = params.info
+    } else {
+      def benchedString = (param.benched ? "Benched " : "")
+      def typeString = (param.hasType ? "${params.hasType} " : "")
+
+      def pokeString = ""
+      if (hasPokeCnt) {
+        i = 1
+        [
+          (params.hasPokemonEX, "Pokémon-EX"),
+          (params.hasPokemonGX, "Pokémon-GX"),
+          (params.hasPokemonV, "Pokémon V"),
+          (params.hasPokemonVMAX, "Pokémon VMAX")
+        ].each{
+          if (it[0]) {
+            pokeString += it[1] + (if (i == isPokeCnt) "" else if (i == isPokeCnt-1) " or " else ", ")
+            i += 1
+          }
+        }
+      } else {
+        pokeString += "Pokémon"
+      }
+
+      failMessage = "${params.opp ? "Your opponent doesn't" : "You don't"} have any ${benchString + typeString + pokeString} that ${params.info ? params.info : "follow the stated condition(s)"}"
```

Comment:
params.info still here...

Reply:
Missed the params.info part earlier... Are you suggesting to move it to it's own variable as with benchString/typeString/pokeString? Or just removing the middle value and make it `params.info ?: "follow[...]"`?

Reply:
just take a look at the beginning of this if/else clause and you'll see why it has to be removed

Reply:
Ahh, ok there was a typo on the if itself. It's meant to be and && not a +.

If params.info is not set, it should generate the pre-formatted failMessage no matter what. If it's set _and_ params.repText is enabled then params.info is used alone, otherwise it replaces the final section of the pre-formatted message.